Subscriber/ApplePayDirectSubscriber.php

```diff
@@ -27,9 +63,82 @@ public static function getSubscribedEvents()
             'Enlight_Controller_Action_PostDispatch_Frontend' => 'onFrontendPostDispatch',
             'Enlight_Controller_Action_PostDispatch_Widget' => 'onFrontendPostDispatch',
             'Enlight_Controller_Action_PostDispatch_Frontend_Checkout' => 'onFrontendCheckoutPostDispatch',
+            'Enlight_Controller_Action_PreDispatch_Frontend_Checkout' => 'writeDefaultPaymentToSession',
+            'Shopware_Modules_Admin_UpdatePayment_FilterSql' => 'onApplePayButtonChangePaymentMethod',
         ];
     }
 
+    public function writeDefaultPaymentToSession(\Enlight_Event_EventArgs $args)
+    {
+        if (
+            !\in_array(
+                $this->adminModule->sGetUserData()['additional']['payment']['name'],
+                [self::APPLEPAY_NAME, self::APPLEPAYDIRECT_NAME],
+                true
+            )
+        ) {
+            return;
+        }
+
+        /** @var Enlight_Controller_Request_RequestHttp $request */
+        $request = $args->get('request');
+
+        if ($request->getActionName() !== 'finish') {
+            return;
+        }
+
+        $userId = $this->adminModule->sGetUserData()['additional']['user']['id'];
+
+        $customer = $this->modelManager->find(Customer::class, $userId);
+
+        $sql = $this->modelManager->getConnection()->prepare(
```

Comment:
please no sql in here, encapsulate it to the corresponding tier, thx

Reply:
Done